src/Raven.Studio/typescript/models/database/tasks/migrateDatabaseModel.ts

```diff
@@ -0,0 +1,58 @@
+﻿/// <reference path="../../../../typings/tsd.d.ts"/>
+
+class migrateDatabaseModel {
+    serverUrl = ko.observable<string>();
+    databaseName = ko.observable<string>();
+    userName = ko.observable<string>();
+    password = ko.observable<string>();
+    domain = ko.observable<string>();
+
+    validationGroup: KnockoutValidationGroup;
+
+    constructor() {
+        this.initValidation();
+    }
+
+    toDto(): Raven.Server.Smuggler.Migration.SingleDatabaseMigrationConfiguration {
+        return {
+            ServerUrl: this.serverUrl(),
+            DatabaseName: this.databaseName(),
+            UserName: this.userName(),
+            Password: this.password(),
+            Domain: this.domain()
+        };
+    }
+
+    private initValidation() {
+        const urlError = ko.observable<string>();
+        this.serverUrl.extend({
+            required: true,
+            validation: [
+                {
+                    validator: (nodeUrl: string) => {
```

Comment:
code duplication, please introduce rule for this

Reply:
done